src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,109 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { ChangeEvent, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  filter: (value : Todo[]) => void,
+};
+
+enum Status {
+  all,
+  active,
+  complete,
+}
+
+export const TodoFilter: React.FC<Props> = ({ todos, filter }) => {
+  const [status, setStatus] = useState(Status.all);
+  const [quote, setQuote] = useState('');
+
+  const handleFilter = (event: ChangeEvent<HTMLSelectElement>
+  | ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuote(value);
+  };
+
+  const handleStatus = (value: Status) => {
+    setStatus(value);
+  };
+
+  useEffect(() => {
+    let filtred = [...todos];
+
+    if (status !== Status.all) {
+      filtred = filtred
+        .filter(todo => {
+          const { completed } = todo;
+
+          switch (status) {
+            case Status.active:
+              return !completed;
+            case Status.complete:
+              return completed;
+            default:
+              return false;
+          }
+        });
+    }
+
+    const lowerQuote = quote.toLowerCase();
+
+    if (quote) {
+      filtred = filtred.filter(todo => {
+        return todo.title.includes(lowerQuote);
+      });
+    }
```

Comment:
now if you have a status not `all` and some quote, you filter todos double times 🥲
maybe you can find a solution how filter just ones )